lib/services/api/session_manager.ts

```diff
@@ -166,7 +166,13 @@ export class SNSessionManager extends PureService<SessionEvent> {
             const email = challengeResponse.values[0].value as string;
             const password = challengeResponse.values[1].value as string;
             const currentKeyParams = await this.protocolService.getAccountKeyParams();
-            const signInResult = await this.signIn(email, password, false, currentKeyParams?.version);
+            const signInResult = await this.signIn(
+              email,
+              password,
+              false,
+              false,
```

Comment:
Hmm, so this is tricky. The function `reauthenticateInvalidSession` is called when the server reports the token is invalid. In the case of an ephemeral session, the server will probably report the token is invalid, then the client will display this. The user will likely enter their credentials, and now, their session is no longer ephemeral.

A solution here could be to determine whether our last valid session was ephemeral, and to use that value here instead of false.

Reply:
Should the sync server be the one to report if a session is ephemeral then? And that's what SNJS would refer to in this case.

Reply:
I can return a property `ephemeral: true` on the session object if you'd like ?

Reply:
Per our private discussion I've gone with the "client remembers if session is ephemeral or not" approach instead